lib/activerecord/delay_touching.rb

```diff
@@ -6,9 +6,9 @@ module DelayTouching
     extend ActiveSupport::Concern
 
     # Override ActiveRecord::Base#touch.
-    def touch(name = nil)
+    def touch(*names)
       if self.class.delay_touching? && !try(:no_touching?)
-        DelayTouching.add_record(self, name)
+        DelayTouching.add_record(self, names)
```

Comment:
the normalize method actually threw me off. But so strong preference either way. 
Another option might also be to keep the add_record method as is and do the loop here
(names || [nil]).each { |name| DelayTouching.add_record(self, name)}